### ISSUE TITLE
Should the front end React application and the backend Vertx service be served from the same port?

### ISSUE THREAD

**Issue by MintyMods** [YoannKy/livi-api#1]
Hi @YoannKy / @jdmnd, 

I found this repo while researching for a job interview that I have upcoming.

I will admit I have no experience with React or Vertx so forked this repo mainly just to learn the stack etc. but I am a little confused over the actual requirements of the task? I wasted a good 30 mins to understand that a separate localhost server listening on port 3000 (determined by the CORS fix) whereas the actual backend server is running separately on port 8080 by default and allowing cross origin requests from port 3000. 

Can I just confirm that the task requires remote access to the backend as there is no mention of running a separate server in the readme.md and I would gladly document this fact for any future potential developers taking the same challenge in the limited 4 hours provided (especially from someone coming from no React or VertX background) and create a Pull Request with said changes.  

My main issue is that I wasted so much time trying to get the front end 'in a browser' after already getting the Java Backend working (i.e. I knew the issue was with the routing to webroot etc) and could have fixed this with a simple new route to serve static resouces from webroot within Vertx etc. but this would have been a different port?

Being a **test** I was trying to stay within the initial compounds of the initial readme.md and the build that was provided which I think can be improved.

Basically I could have done either route but spent soo much time trying to work out the 'potential bug' which seems to be with the documentation I had very little time to fix the issue and agree this is mostly due to my lack of React skills but just wanted to point out the documentation could be improved to at least allow a fair test for those interested.

**Comment by MintyMods** [comment]
I will say I had fun learning React but it is not as good as [Svelte](http://svelte.dev/) and after my initial 4 hours I only really produced the following - https://github.com/MintyMods/livi-api/pull/1/commits/e462baeed83e190c99cc542e776e87403037ef4f but the [Vertx](https://vertx.io/) I find very interesting, I could have used that with my [last personal project](https://github.com/MintyMods/MintyHydro) had I known about it.
